exercise/tests.py

```diff
@@ -587,15 +587,22 @@ def test_unofficial_submission(self):
         self.learning_object_category.accept_unofficial_submits = True
         self.learning_object_category.save()
 
+        self.submission_when_late_allowed.set_error()
+        self.late_late_submission_when_late_allowed.set_error()
+        self.submission_when_late_allowed.save()
+        self.late_late_submission_when_late_allowed.save()
+
         self.late_submission_when_late_allowed.set_points(10, 10)
         self.late_submission_when_late_allowed.set_ready()
         self.late_submission_when_late_allowed.save()
         self.assertEqual(self.late_submission_when_late_allowed.grade, 100)
         self.assertEqual(self.late_submission_when_late_allowed.status, Submission.STATUS.UNOFFICIAL)
         summary = UserExerciseSummary(self.base_exercise_with_late_submission_allowed, self.user)
-        self.assertEqual(summary.get_submission_count(), 3)
+        self.assertEqual(summary.get_submission_count(), 1)
         self.assertEqual(summary.get_points(), 0) # unofficial points are not shown here
         self.assertFalse(summary.is_graded())
+        # 3 submissions:
+        # initialized, unofficial, initialized
         self.assertTrue(summary.is_unofficial())
```

Comment:
In this test, there were 3 submissions with the states INITIALIZED, UNOFFICIAL, INITIALIZED. `assertTrue` on line 599/606 would fail because the initialized submissions can now "de-unofficialize" the summary. To be in line with the test's original intentions, I decided to set the states of the INITIALIZED submissions to ERROR, so the assertion would succeed. Perhaps not the cleanest way to update this test? It was broken by the earlier commit.

Reply:
The comments on lines 604-605 are temporary (this is a draft PR). But perhaps these tests could use some extra comments? Without any comments it's pretty annoying to figure out what submissions there are and what is being tested at which point.

Reply:
Yes, and I could write a couple more comments to further describe what the test is supposed to test.